server/src/main/java/org/elasticsearch/ingest/ConditionalProcessor.java

```diff
@@ -39,16 +35,6 @@
  */
 public class ConditionalProcessor extends AbstractProcessor implements WrappingProcessor {
 
-    private static final DeprecationLogger deprecationLogger = DeprecationLogger.getLogger(DynamicMap.class);
-    private static final Map<String, Function<Object, Object>> FUNCTIONS = Map.of("_type", value -> {
-        deprecationLogger.warn(
-            DeprecationCategory.INDICES,
-            "conditional-processor__type",
-            "[types removal] Looking up doc types [_type] in scripts is deprecated."
-        );
-        return value;
-    });
```

Comment:
Is there a reason we wouldn't do:

```java
FUNCTIONS = Map.of("_type", ignored -> "_doc");
```

and just hardcode it to return `_doc`?

Reply:
I haven't found a PR that clearly is the one that makes the _type be null for these, but I have confirmed that if you run 8.0.0 the behavior is the same as I show above. So this has been the behavior since at least February 2022.